frontend/components/Header.tsx

```diff
@@ -2,26 +2,39 @@ import * as React from 'react';
 import {
 	Link,
 } from 'react-router-dom';
+import { connect } from 'react-redux';
+// eslint-disable-next-line no-unused-vars
+import {AppStore} from '../storage/common';
 
-export class Header extends React.Component {
-	public render(): JSX.Element {
-		return (
-			<>
-				<nav>
-					<div><strong>Metroidvania Compendium</strong>
-						<span> » Home</span>
-					</div>
-					<div>
-						{/* Spans are placeholder before styling, to have a space between links */}
-						<Link to="/">Home</Link><span>  </span>
-						<Link to="/games">Game List</Link><span>  </span>
-						<Link to="/abilities">Abilities</Link><span>  </span>
-						<Link to="/config">Config</Link><span>  </span>
-					</div>
-				</nav>
-				<hr></hr>
-			</>
-
-		);
-	}
+interface HeaderProps{
+	currBreadcrumb: AppStore | undefined;
 }
+const Header: React.FC<HeaderProps> = (props) => {
+	const breadcrumb = props.currBreadcrumb.headerBreadcrumb || 'Missing page name';
+	console.log(breadcrumb);
+	return (
+		<>
+			<nav>
+				<div><strong>Metroidvania Compendium</strong>
+					<span> » {breadcrumb}</span>
+				</div>
+				<div>
+					{/* Spans are placeholder before styling, to have a space between links */}
+					<Link to="/">Home</Link><span>  </span>
+					<Link to="/games">Game List</Link><span>  </span>
+					<Link to="/abilities">Abilities</Link><span>  </span>
+					<Link to="/config">Config</Link><span>  </span>
+				</div>
+			</nav>
+			<hr></hr>
+		</>
+
+	);
+};
+
+const mapPropsToStore = (state: AppStore): HeaderProps => {
+	return {
+		currBreadcrumb: state ? state : undefined,
```

Comment:
Typescript guarantees `AppStore` is set, so you don't need to do the ternary operator check, you could just set it to `state`...

... expect an earlier comment says to make it a string, so directly extract the breadcrumb from the store here and if you don't have a strong guarantee in the store that the value is not-null, add `... || ''` at the end.